pkg/crc/config/viper_config.go

```diff
@@ -180,20 +143,29 @@ func Set(key string, value interface{}) (string, error) {
 }
 
 func (c *Config) Set(key string, value interface{}) (string, error) {
-	_, ok := c.allSettings[key]
+	setting, ok := c.allSettings[key]
 	if !ok {
 		return "", fmt.Errorf(configPropDoesntExistMsg, key)
 	}
 
-	ok, expectedValue := c.allSettings[key].validationFn(value)
+	var castValue interface{}
+	switch setting.defaultValue.(type) {
+	case int:
+		castValue = cast.ToInt(value)
+	case string:
+		castValue = cast.ToString(value)
+	case bool:
+		castValue = cast.ToBool(value)
+	}
```

Comment:
Now that I look closer at this casting, I wonder if we should be checking/reporting errors?

Reply:
I added one commit: https://github.com/code-ready/crc/pull/1460/commits/985f58c3f99a8621bcd0ba2349b0f62c06eec56f